src/sources/opentelemetry/http.rs

```diff
@@ -103,12 +114,69 @@ pub(crate) fn build_warp_filter(
     log_filters.or(trace_filters).unify().boxed()
 }
 
+fn enrich_events(
+    events: &mut [Event],
+    headers: &Vec<HttpConfigParamKind>,
+    headers_config: &HeaderMap,
+    log_namespace: LogNamespace,
+) {
+    for event in events.iter_mut() {
+        match event {
+            Event::Log(log) => {
+                for h in headers {
+                    match h {
+                        // Add each non-wildcard containing header that was specified
+                        // in the `headers` config option to the event if an exact match
+                        // is found.
+                        HttpConfigParamKind::Exact(header_name) => {
+                            let value = headers_config.get(header_name).map(HeaderValue::as_bytes);
+
+                            log_namespace.insert_source_metadata(
+                                SimpleHttpConfig::NAME,
+                                log,
+                                Some(LegacyKey::InsertIfEmpty(path!(header_name))),
+                                path!("headers", header_name),
+                                Value::from(value.map(Bytes::copy_from_slice)),
+                            );
+                        }
+                        // Add all headers that match against wildcard pattens specified
+                        // in the `headers` config option to the event.
+                        HttpConfigParamKind::Glob(header_pattern) => {
+                            for header_name in headers_config.keys() {
+                                if header_pattern.matches_with(
+                                    header_name.as_str(),
+                                    glob::MatchOptions::default(),
+                                ) {
+                                    let value =
+                                        headers_config.get(header_name).map(HeaderValue::as_bytes);
+
+                                    log_namespace.insert_source_metadata(
+                                        SimpleHttpConfig::NAME,
```

Comment:
Replace with `OpentelemetryConfig` (or this can also resolved by the refactoring I suggested above).